torch_geometric/nn/models/explainer.py

```diff
@@ -144,3 +149,213 @@ def to_captum(model: torch.nn.Module, mask_type: str = "edge",
             the index specified. (default: :obj:`None`)
     """
     return CaptumModel(model, mask_type, output_idx)
+
+
+class Explainer(torch.nn.Module):
+    r"""Base Explainer class for Graph Neural Networks, e.g.
+    PGExplainer and GNNExplainer."""
+
+    # TODO: Change docstring
+    # TODO: Rearange args
+    def __init__(self, model: torch.nn.Module, lr: Optional[float] = None,
+                 epochs: Optional[int] = None, num_hops: Optional[int] = None,
+                 return_type: str = 'log_prob', log: bool = False, **kwargs):
+        super().__init__()
+        assert return_type in ['log_prob', 'prob', 'raw', 'regression']
+
+        self.model = model
+        self.lr = lr
+        self.epochs = epochs
+        self.num_hops = num_hops
+        self.return_type = return_type
+        self.log = log
+
+    @property
+    def num_hops(self):
+        return self._num_hops
+
+    @num_hops.setter
+    def num_hops(self, num_hops):
+        if num_hops is not None:
+            self._num_hops = num_hops
+        else:
+            # Could this be a general utils function?
+            k = 0
+            for module in self.model.modules():
+                if isinstance(module, MessagePassing):
+                    k += 1
+            self._num_hops = k
+
+    def __flow__(self):
+        for module in self.model.modules():
+            if isinstance(module, MessagePassing):
+                return module.flow
+        return 'source_to_target'
+
+    # TODO: Find out, whether to change variable name node_idx here as well?
+    def __subgraph__(self, node_idx, x, edge_index, **kwargs):
```

Comment:
```suggestion
    def subgraph(self, node_idx, x, edge_index, **kwargs):
```
Let's expose this function.